whois-nrtm4/src/main/java/net/ripe/db/nrtm4/GzipOutStreamWriter.java

```diff
@@ -16,30 +16,40 @@ public class GzipOutStreamWriter {
     final ByteArrayOutputStream bos;
     final GZIPOutputStream gzOut;
 
-    public GzipOutStreamWriter() throws IOException {
-        this.bos = new ByteArrayOutputStream();
-        this.gzOut = new GZIPOutputStream(bos);
+    public GzipOutStreamWriter() {
+        try {
+            this.bos = new ByteArrayOutputStream();
+            this.gzOut = new GZIPOutputStream(bos);
+        } catch (IOException e) {
+            LOGGER.warn("Error while creating outputstream", e);
+            throw new RuntimeException(e);
+        }
     }
 
-    public void write(final NrtmFileRecord record) throws IOException {
-        gzOut.write(NrtmFileUtil.convertToJSONTextSeq(record).getBytes());
+    public void write(final NrtmFileRecord record)  {
+        try {
+            gzOut.write(NrtmFileUtil.convertToJSONTextSeq(record).getBytes());
+        } catch (IOException e) {
+            LOGGER.warn("Error while writing snapshotfile", e);
+            throw new RuntimeException(e);
+        }
     }
 
     public ByteArrayOutputStream getOutputstream() {
         return this.bos;
     }
 
     public void close() {
-        LOGGER.info("closing resources");
+        LOGGER.debug("closing resources");
         try {
             gzOut.close();
         } catch (IOException e) {
-            LOGGER.error("Exception while closing gzipStream {}", e);
+            LOGGER.debug("Exception while closing gzipStream {}", e);
         }
         try {
             bos.close();
         } catch (IOException e) {
-            LOGGER.error("Exception while closing outputstream {}", e);
+            LOGGER.debug("Exception while closing outputstream {}", e);
```

Comment:
This should be logged as Error, not as debug, right?

Reply:
So that when it tries to close already closed stream it will not log it as error . Also you cannot read the bytes from open output stream so it will throw a runtime error which will be logged